components/engine_traits/src/tablet.rs

```diff
@@ -275,13 +271,16 @@ impl<EK> TabletRegistry<EK> {
         }
         let tablet = self.tablets.factory.open_tablet(ctx, &path)?;
         let mut cached = self.get_or_default(id);
-        cached.set(tablet);
+        let old_tablet = cached.set(tablet);
```

Comment:
You can't just put all these tablets into tombstone records. For example, in split/merge, the original tablets can't be gced until admin.persisted_applied is advanced.

Reply:
I updated the code to hold an extra reference inside peer, the reference is dropped after new_tablet is flushed.